src/components/Select/Select.tsx

```diff
@@ -0,0 +1,271 @@
+        },
+        ref: Ref<HTMLDivElement>
+    ) => {
+        const [dropdownVisible, setDropdownVisibility] =
+            useState<boolean>(false);
+        const [options, setOptions] = useState<SelectOption[]>(
+            _options.map((option) => ({
+                selected: false,
+                hideOption: false,
+                ...option,
+            }))
+        );
+        const [searchQuery, setSearchQuery] = useState<string>('');
+
+        useEffect(() => {
+            const selected = options.filter((option) => option.selected);
+            setOptions(
+                _options.map((option) => ({
+                    selected: !!selected.find(
+                        (opt) => opt.value === option.value
+                    ),
+                    hideOption: false,
+                    ...option,
+                }))
+            );
+        }, [isLoading]);
+
+        useEffect(() => {
+            const updatedOptions = options.map((opt) => ({
+                ...opt,
+                selected:
+                    (defaultValue && opt.value === defaultValue) ||
+                    opt.selected,
+            }));
+            setOptions(updatedOptions);
+        }, [defaultValue]);
+
+        const toggleOption = (option: SelectOption) => {
+            setSearchQuery('');
+            const updatedOptions = options.map((opt) => {
+                const defaultState = multiple ? opt.selected : false;
+                const selected =
+                    opt.value === option.value ? !opt.selected : defaultState;
+                return {
+                    ...opt,
+                    selected: selected,
+                };
+            });
+            setOptions(updatedOptions);
+        };
+
+        const onInputClear = () => {
+            if (filterable && dropdownVisible) {
+                setSearchQuery('');
+            } else {
+                setOptions(
+                    options.map((opt) => ({
+                        ...opt,
+                        selected: false,
+                    }))
+                );
+            }
+        };
+
+        const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+            const { target } = event;
+            const value = target?.value?.toLowerCase();
+            setSearchQuery(value);
+            if (loadOptions) {
+                return loadOptions(value);
+            }
+            if (value) {
+                setOptions(
+                    options.map((option) => ({
+                        ...option,
+                        hideOption: !option.text.toLowerCase().includes(value),
+                    }))
+                );
+            } else {
+                setSearchQuery('');
+                setOptions(
+                    options.map((option) => ({ ...option, hideOption: false }))
+                );
+            }
+        };
+
+        const beforeToggleStateChange = (show: boolean) => {
```

Comment:
nit: return type
this resolve to a boolean, can you change the name to something more apt